### PR TITLE
Import configuration/prepare import op

### PR DESCRIPTION
**Pull request by kaulfield23**
## Description
This PR is testing `prepareImportOperation` function that maps data from configuration step for sending to API

## Changes

* Adds `prepareImportOperation.ts`
* Changes `OrgColumn` type and its relevant code

## Related issues
Part of import-configuration

### CODE REVIEW

File: src/features/import/utils/prepareImportOperation.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import { CellData, ColumnKind, Sheet } from './types';
+
+export type OpsType = {
+  fields?: Record<string, CellData>;
+  op: 'person.import';
+  organizations?: number | null;
+  tags?: number[];
+};
+export interface ImportOpsProp {
```
Comment (ziggabyte) [inline src/features/import/utils/prepareImportOperation.ts:9]: I think the name of this is confusing me a bit. The name sounds as if it is a type of props to a component, but it's used as the type for what this function returns... could ImportOperations be a better name?

File: src/features/import/utils/prepareImportOperation.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  ops: OpsType[];
+}
+
+export function prepareImportOperations(configData: Sheet): ImportOpsProp {
```
Comment (ziggabyte) [inline src/features/import/utils/prepareImportOperation.ts:13]: maybe call this `configuredSheet` instead of `configData`, so it's easy to remember that the type of this object is a `Sheet`?

**Review by ziggabyte — COMMENTED** [review]
Just giving a mini feedback on naming, I can contribute a more substansial review tomorrow if needed :)

File: src/features/import/utils/prepareImportOperation.ts (hunk 1 above)
Comment (kaulfield23) [reply src/features/import/utils/prepareImportOperation.ts:9]: I always add `Prop` as suffix at the end of `interface`. Thought that it is a pattern in our code base?? 🤷 No idea.. Could add `I` as prefix as well

File: src/features/import/utils/prepareImportOperation.ts (hunk 2 above)
Comment (kaulfield23) [reply src/features/import/utils/prepareImportOperation.ts:13]: Yes I agree!

File: src/features/import/utils/prepareImportOperation.ts (hunk 1 above)
Comment (richardolsson) [reply src/features/import/utils/prepareImportOperation.ts:9]: The `Prop` prefix, or really `Props` (plural) is what we use when the interface/type defines the properties (props) for a component. It's not something we _always_ use for _all_ interfaces/types.

We also don't use the `I` prefix, as far as I know, in this codebase. I've used it elsewhere, but I don't think it's used here.

Thinking about it now, I kind of think this function should really just return a list of ops, not an object containing the list of ops, unless you see some reason why I'm wrong about this. So the return value for the function should be `OpsType[]`.

Although I also think the name `OpsType` is not great. Maybe it could be `ZetkinPersonImportOp` instead, to indicate that it's related to the Zetkin API, and that it's a single operation of type `person.import`?

In summary, I think you should:
* Remove the `ImportOpsProp` entirely
* Rename `OpsType` to `ZetkinPersonImportOp`
* Refactor `prepareImportOperations` to return an array of operations instead

File: src/features/import/utils/prepareImportOperation.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import { CellData, ColumnKind, Sheet } from './types';
```
Comment (richardolsson) [inline src/features/import/utils/prepareImportOperation.ts:1]: The name of this file `prepareImportOperation` (singular) does not match the name of the default exported function `prepareImportOperations` (plural, which is correct).

File: src/features/import/utils/prepareImportOperation.ts (hunk 2 above)
Comment (richardolsson) [inline src/features/import/utils/prepareImportOperation.ts:13]: I think this should be the default export.

**Review by richardolsson — COMMENTED** [review]
Some quick comments below.

File: src/features/import/utils/prepareImportOperation.ts (hunk 1 above)
Comment (kaulfield23) [reply src/features/import/utils/prepareImportOperation.ts:9]: Thanks for feedback @ziggabyte and @richardolsson ! I should have thought about naming things more carefully. Now I know so will try to be better next time 👍

**Comment by kaulfield23** [comment]
I try to avoid repeated codes that excludes rows with empty string or null in tags / orgs like the screenshot. But typescript complained about mapping so I decided to write two repeated codes for both tags and orgs. Please let me know if there is a better solution for this!

image

File: src/features/import/utils/prepareImportOperations.spec.ts [hunk 4]
```diff
@@ -0,0 +1,790 @@
+import prepareImportOperations from './prepareImportOperations';
+import { ColumnKind, Sheet } from './types';
+import { describe, it } from '@jest/globals';
+
+describe('prepareImportOperations when first row is header', () => {
```
Comment (richardolsson) [inline src/features/import/utils/prepareImportOperations.spec.ts:5]: I would wrap both this `describe` and the one on line 383 in an outer describe, like this:

```ts
describe('pareImportOperations()', () => {
  describe('when first row is header', () => {
    // it tests
  });

  describe('when first row is not header', () => {
    // it tests
  });
});
```

File: src/features/import/utils/prepareImportOperations.spec.ts [hunk 5]
```diff
@@ -0,0 +1,790 @@
...
+  it('converts ID only', () => {
+    const configData: Sheet = {
+      columns: [
+        { idField: 'ext_id', kind: ColumnKind.ID_FIELD, selected: true },
```
Comment (richardolsson) [inline src/features/import/utils/prepareImportOperations.spec.ts:9]: I would also like a test that handles the "Zetkin ID" type of ID column.

**Review by richardolsson — CHANGES_REQUESTED** [review]
I think this looks good. I do think there are unnecessarily many tests. I think the following tests would be enough, IMO:

* Handles external ID column (test with single column)
* Handles Zetkin ID column (test with single column)
* Handles field columns (test with a few different field columns)
* Handles tag columns (test with a few tag columns)
* Handles org columns (test with few org columns)
* Ignores unknown columns (test with some "real", and one unknown)
* Ignores first row when first row is header

The goal is to test all aspects of the function, not to test every single combination of things. But I see no reason to change it now. It would just have saved some time during development.

But I'm missing a test for the Zetkin ID type of ID column. Can that be added?

**Review by richardolsson — APPROVED** [review]
LGTM! 💯
